MasterServer/Program.cs

```diff
@@ -6,114 +6,146 @@
 
 namespace MasterServer
 {
-	class Program
+	internal static class Program
 	{
+		private static bool s_Daemon = false;
+		private static string s_Host = "0.0.0.0";
+		private static ushort s_Port = 15940;
+		private static int s_EloRange = 0;
+
+		private static MasterServer s_Server;
+
 		private static void Main(string[] args)
 		{
-			string host = "0.0.0.0";
-			ushort port = 15940;
-			string read = string.Empty;
-			int eloRange = 0;
+			ParseArgs(args);
+
+			Console.WriteLine("Hosting ip [{0}] on port [{1}]", s_Host, s_Port);
+			PrintHelp();
+
+			s_Server = new MasterServer(s_Host, s_Port)
+			{
+				EloRange = s_EloRange
+			};
+			s_Server.ToggleLogging();
+
+			while (true)
+			{
+				if (!s_Daemon)
+					HandleConsoleInput();
+			}
+		}
 
+		private static void ParseArgs(string[] args)
+		{
 			Dictionary<string, string> arguments = ArgumentParser.Parse(args);
 
 			if (args.Length > 0)
 			{
-				if (arguments.ContainsKey("host"))
-					host = arguments["host"];
+				string value;
+				if (arguments.TryGetValue("d", out value) || arguments.TryGetValue("daemon", out value))
+					s_Daemon = true;
 
-				if (arguments.ContainsKey("port"))
-					ushort.TryParse(arguments["port"], out port);
+				if (arguments.TryGetValue("h", out value) || arguments.TryGetValue("host", out value))
+					s_Host = value;
 
-				if (arguments.ContainsKey("elorange"))
-					int.TryParse(arguments["elorange"], out eloRange);
+				if (arguments.TryGetValue("p", out value) || arguments.TryGetValue("port", out value))
+					ushort.TryParse(value, out s_Port);
+
+				if (arguments.TryGetValue("e", out value) || arguments.TryGetValue("elorange", out value))
+					int.TryParse(value, out s_EloRange);
 			}
 			else
 			{
 				Console.WriteLine("Entering nothing will choose defaults.");
-				Console.WriteLine("Enter Host IP (Default: "+ GetLocalIPAddress() + "):");
-				read = Console.ReadLine();
-				if (string.IsNullOrEmpty(read))
-					host = GetLocalIPAddress();
-				else
-					host = read;
+				Console.WriteLine("Enter Host IP (Default: " + GetLocalIpAddress() + "):");
+				string read = Console.ReadLine();
+				s_Host = string.IsNullOrEmpty(read) ? GetLocalIpAddress() : read;
 
 				Console.WriteLine("Enter Port (Default: 15940):");
 				read = Console.ReadLine();
 				if (string.IsNullOrEmpty(read))
-					port = 15940;
+					s_Port = 15940;
 				else
-				{
-					ushort.TryParse(read, out port);
-				}
+					ushort.TryParse(read, out s_Port);
 			}
+		}
 
-			Console.WriteLine(string.Format("Hosting ip [{0}] on port [{1}]", host, port));
-			PrintHelp();
-			MasterServer server = new MasterServer(host, port);
-			server.EloRange = eloRange;
-			server.ToggleLogging();
+		private static void HandleConsoleInput()
+		{
+			string read = Console.ReadLine()?.ToLower();
```

Comment:
Not sure here, does this operator work on .Net 3.5?

Reply:
Aha, no it does not, I wasn't clear on which version we are using and it seemed to compile/run fine for me. Will fix.